build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/BuildView.java

```diff
@@ -107,6 +115,21 @@ public String description() {
         return getOrElse(build.getDescription(), "");
     }
 
+    @Override
+    public List<String> stages() {
+        if (this.build instanceof WorkflowRun) {
+            WorkflowRun currentBuild = (WorkflowRun) this.build;
+
+            if (currentBuild.isBuilding()) {
+                StageTraversal traversal = new StageTraversal();
+                traversal.start(currentBuild.getExecution().getCurrentHeads());
```

Comment:
If you check the source for WorkflowRun.java you see that there's a findbugs annotation `@CheckForNull`  on getExecution(), so apparently it's supposed to return null occasionally.

https://github.com/jenkinsci/workflow-job-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L660

There's also the following comment:
~~~
/**
 * Gets the associated execution state.
 * @return non-null after the flow has started, even after finished (but may be null temporarily when about to start, or if starting failed)
 */
~~~
So it could apparently return null at start.